hazelcast-client/src/main/java/com/hazelcast/client/spi/ProxyManager.java

```diff
@@ -197,6 +200,12 @@ public ClientProxy create(String id, ClientContext context) {
                 return new ClientIdGeneratorProxy(IdGeneratorService.SERVICE_NAME, id, context, atomicLong);
             }
         });
+        register(FlakeIdGeneratorService.SERVICE_NAME, new ClientProxyFactoryWithContext() {
```

Comment:
It looks like you can directly use
`register(FlakeIdGeneratorService.SERVICE_NAME, FlakeIdGeneratorService.class)`
No need for special factory.